include/moqt/core/base_track_handler.h

```diff
@@ -0,0 +1,138 @@
+/*
+ *  Copyright (C) 2024
+ *  Cisco Systems, Inc.
+ *  All Rights Reserved
+ */
+#pragma once
+
+#include <optional>
+#include <transport/span.h>
+#include <moqt/common.h>
+#include <vector>
+
+namespace moq::transport {
+
+    /**
+     * @brief MoQ track base handler for tracks (subscribe/publish)
+     *
+     * @details Base MoQ track handler
+     */
+    class BaseTrackHandler
+    {
+      public:
+        friend class Transport;
+
+        /**
+         * @brief Track mode object of object published or received
+         *
+         * @details QUIC stream handling mode used to send objects or how object was received
+         */
+        enum class TrackMode : uint8_t
+        {
+            kDatagram,
+            kStreamPerObject,
+            kStreamPerGroup,
+            kStreamPerTrack
+        };
+
+        virtual ~BaseTrackHandler() = default;
+
+        // --------------------------------------------------------------------------
+        // Public API methods that normally should not be overridden
+        // --------------------------------------------------------------------------
+
+        BaseTrackHandler() = delete;
+
+      protected:
+        /**
+         * @brief Track delegate constructor
+         *
+         * @param full_track_name       Full track name struct
+         */
+        BaseTrackHandler(const FullTrackName& full_track_name)
+          : full_track_name_(full_track_name)
+        {
+        }
+
+        // --------------------------------------------------------------------------
+        // Public Virtual API callback event methods to be overridden
+        // --------------------------------------------------------------------------
+      public:
+        /**
+         * @brief Set the track alias
+         * @details MOQ transport instance will set the track alias when the track has
+         *   been assigned.
+         *
+         * @param track_alias       MoQT track alias for track namespace+name that
+         *                          is relative to the QUIC connection session
+         */
+        void SetTrackAlias(uint64_t track_alias) { full_track_name_.track_alias = track_alias; }
+
```

Comment:
we should add a note that this api will override the existing value. Also is it OK to call this api. mid-sessopn ?